src/main/scala/za/co/absa/standardization/schema/StdSchemaUtils.scala

```diff
@@ -46,10 +51,32 @@ object StdSchemaUtils {
         .replace('.', '_')
   }
 
+  def evaluateColumnName(columnName: String): Column = {
+    def segmentToColumn(colFnc: String => Column, columnSegment: String): Column = {
+      val PatternForSubfield = """^(.+)\[(.+)]$""".r
```

Comment:
```suggestion
      val PatternForSubfield = """^(.+)\[(.+)\]$""".r
```
❓

Reply:
`]` has to be escaped only inside the character classes.

Reply:
Both seem to be working, but the one proposed by @lsulak seems more natural to me. Actually I'm surprised that `]` doesn't have to be escaped.

Reply:
If I understood you correctly, then there is still one improvement possible (thanks for the explanation Sasa about the character classes - it makes sense):

```suggestion
      val PatternForSubfield = """^(.+)[(.+)]$""".r
```

Exactly, why to complicate it by having it for open bracket and not for the closing one?

Reply:
Well the first escape is necessary I believe. Otherwise `[` will act as start of list characters to match instead of the literal `[`. 🤔

Reply:
Your arguments & facts I think more than satisfied (i.e. destroyed :D ) my original questions. Makes sense, of course!